app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,226 @@
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+    }
+
+    @Nested
+    @DisplayName("getTasks 메소드는")
+    class Describe_getTasks {
+        List<Task> subject() {
+            return taskService.getTasks();
+        }
+
+        @Nested
+        @DisplayName("DB에 데이터가 존재하지 않는다면")
+        class Context_with_empty_db {
+            @Test
+            @DisplayName("빈 ArrayList를 리턴한다")
+            void it_returns_empty_arraylist() {
+                List<Task> findTasks = subject();
+
+                assertThat(findTasks).hasSize(0);
+                assertThat(findTasks).isEqualTo(new ArrayList<>());
+            }
+        }
+
+        @Nested
+        @DisplayName("DB에 데이터가 존재한다면")
+        class Context_with_non_empty_db extends NewTask {
+            private Task task1, task2;
+            private int expectedSize;
+
+            @BeforeEach
+            void prepareTests() {
+                task1 = withTitle("할 일 추가1");
+                task2 = withTitle("할 일 추가2");
+
+                taskService.createTask(task1);
+                taskService.createTask(task2);
+
+                expectedSize = taskService.getTasks().size();
+            }
+
+            @Test
+            @DisplayName("존재하는 모든 할 일을 리턴한다")
+            void it_returns_all_tasks() {
+                List<Task> findTasks = subject();
+                final int actualSize = findTasks.size();
+
+                assertThat(actualSize).isEqualTo(expectedSize);
+                assertThat(findTasks.get(0).getTitle()).isEqualTo(task1.getTitle());
+                assertThat(findTasks.get(1).getTitle()).isEqualTo(task2.getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_createTask {
+        @Nested
+        @DisplayName("새로운 할 일이 주어지면")
+        class Context_with_new_task extends NewTask {
+            private final Long TASK_ID = 1L;
+            private final String TITLE = "할 일 추가";
+            private final Task givenTask = withTitle(TITLE);
+            private int expectedSize;
+
+            @BeforeEach
+            void prepareTest() {
+                expectedSize = taskService.getTasks().size() + 1;
+            }
+
+            @Test
+            @DisplayName("DB에 등록하고 등록된 할 일을 리턴한다")
+            void it_returns_registered_task() {
+                Task actualTask = taskService.createTask(givenTask);
+
+                final int actualSize = taskService.getTasks().size();
+                final Task actualFindTask = taskService.getTask(TASK_ID);
+
+                assertThat(actualSize).isEqualTo(expectedSize);
+                assertThat(actualTask.getTitle()).isEqualTo(givenTask.getTitle());
+                assertThat(actualFindTask.getTitle()).isEqualTo(givenTask.getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Nested
+        @DisplayName("검색한 할 일 ID가 존재하면")
```

Comment:
이렇게 해보면 어떨까요?

```suggestion
        @DisplayName("ID가 주어지면")
```

다음 두 문장을 비교해가며 여러 차례 소리내서 읽어보세요.

- getTask 메소드는, ID가 주어지면, 할 일을 찾아 리턴한다.
- getTask 메소드는, 검색한 할 일 ID가 존재하면, 해당 ID의 할 일을 리턴한다.

Reply:
다시 한번 수정해봤습니다~